src/Microsoft.AspNetCore/CertificateLoader.cs

```diff
@@ -12,23 +12,90 @@ namespace Microsoft.AspNetCore
     /// <summary>
     /// A helper class to load certificates from files and certificate stores based on <seealso cref="IConfiguration"/> data.
     /// </summary>
-    public static class CertificateLoader
+    public class CertificateLoader
     {
+        private readonly IConfiguration _certificatesConfiguration;
+
+        /// <summary>
+        /// Creates a new instance of <see cref="KestrelServerOptionsSetup"/>.
+        /// </summary>
+        public CertificateLoader()
+        {
+        }
+
+        /// <summary>
+        /// Creates a new instance of <see cref="KestrelServerOptionsSetup"/> that can load certificates by name.
+        /// </summary>
+        /// <param name="certificatesConfig">An <see cref="IConfiguration"/> instance with information about certificates.</param>
+        public CertificateLoader(IConfiguration certificatesConfig)
+        {
+            _certificatesConfiguration = certificatesConfig;
+        }
+
+        /// <summary>
+        /// Loads one or more certificates based on the information found in a configuration section.
+        /// </summary>
+        /// <param name="certificateConfiguration">A configuration section containing either a string value referencing certificates
+        /// by name, or one or more inline certificate specifications.
+        /// </param>
+        /// <returns>One or more loaded certificates.</returns>
+        public IEnumerable<X509Certificate2> Load(IConfigurationSection certificateConfiguration)
```

Comment:
By "keys" do you mean the certificate name used in config? Yeah, I think it would be good to provide those.